### PR TITLE
Fix URL table for ingresses

### LINKED ISSUE #1307: Ingress URL not shown in the AppView
**Description**

URL of the ingress are not shown in the Access URL table

**Steps to reproduce the issue:**

1. Deploy Wordpress
2. Enable the Ingress rule

**Additional information you deem important (e.g. issue happens only occasionally):**

Apparently, the `AccessURLTable` is being mounted before the application manifests is parsed. It's only when mounting it when the component `AccessURLTable` fetches the existing ingress and at that point the list is empty.

**Version of Helm, Kubeapps and Kubernetes**:

- Output of `helm version`:

```
Server: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
```

- Output of `helm list <kubeapps-release-name>`:

```
kubeapps	1       	Wed Nov 20 15:29:47 2019	DEPLOYED	kubeapps-3.0.3v1.6.1     	kubeapps 
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:23:11Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.8-gke.17", GitCommit:"188432a69210ca32cafded81b4dd1c063720cac0", GitTreeState:"clean", BuildDate:"2019-11-13T20:47:11Z", GoVersion:"go1.12.11b4", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by andresmgot**
**Description of the change**

The component `AccessURLTable` is now being mounted *before* the references to the ingress objects are populated. This caused the AccessURLTable to be empty. To avoid that, we are now fetching also the ingresses when there are changes in the list.

Another subtle change is that instead of relying on the implementation of `fetchIngresses`, which was defined in the container, I have moved the implementation to the component so it's clear what it's doing.

**Applicable issues**

Fixes #1307

### CODE REVIEW

File: dashboard/src/components/AppView/AccessURLTable/AccessURLTable.test.tsx [hunk 1]
```diff
@@ -4,15 +4,32 @@ import * as React from "react";
...
+  getResource: jest.fn(),
+};
+
+describe("when receiving ingresses", () => {
+  it("fetches ingresses at mount time", () => {
+    const mock = jest.fn();
+    shallow(
+      <AccessURLTable {...defaultProps} ingressRefs={[{} as ResourceRef]} getResource={mock} />,
+    );
+    expect(mock).toHaveBeenCalled();
```
Comment (absoludity) [inline dashboard/src/components/AppView/AccessURLTable/AccessURLTable.test.tsx]: Shouldn't this be `.toHaveBeenCalledWith(...)` to be able to justify the name of the test? (ie. to show that it fetches ingresses at mount time)

File: dashboard/src/components/AppView/AccessURLTable/AccessURLTable.test.tsx [hunk 2]
```diff
@@ -4,15 +4,32 @@ import * as React from "react";
...
+  });
+
+  it("fetches new ingresses when received", () => {
```
Comment (absoludity) [inline dashboard/src/components/AppView/AccessURLTable/AccessURLTable.test.tsx]: s/when received/when new ingress refs received/ (just to be specific.

File: dashboard/src/components/AppView/AccessURLTable/AccessURLTable.test.tsx [hunk 3]
```diff
@@ -4,15 +4,32 @@ import * as React from "react";
...
     const mock = jest.fn();
-    shallow(<AccessURLTable services={[]} ingresses={[]} fetchIngresses={mock} />);
+    const wrapper = shallow(<AccessURLTable {...defaultProps} getResource={mock} />);
+    wrapper.setProps({ ingressRefs: [{} as ResourceRef] });
     expect(mock).toHaveBeenCalled();
```
Comment (absoludity) [inline dashboard/src/components/AppView/AccessURLTable/AccessURLTable.test.tsx]: Ditto here. I realise this line didn't change, but it was previously a mock of the function `fetchIngresses` so it was obvious. now it's a mock of the more generic `getResource`, so it'd be clearer to show that it is indeed getting an Ingress resource.

File: dashboard/src/components/AppView/AccessURLTable/AccessURLTable.tsx [hunk 4]
```diff
@@ -90,6 +96,12 @@ class AccessURLTable extends React.Component<IAccessURLTableProps> {
     }
     return;
   }
+
+  private fetchIngresses() {
+    // Fetch all related Ingress resources. We don't need to fetch Services as
+    // they are expected to be watched by the ServiceTable.
+    this.props.ingressRefs.forEach(r => this.props.getResource(r));
```
Comment (absoludity) [inline dashboard/src/components/AppView/AccessURLTable/AccessURLTable.tsx]: We don't need to worry about re-fetching ingresses we've previously fetched - I assume?

Comment (andresmgot) [reply dashboard/src/components/AppView/AccessURLTable/AccessURLTable.tsx]: not really, it will just update the info

**Pull request merged**